web3/inpage.js

```diff
@@ -174,14 +49,11 @@ if (!window.chrome || !window.chrome.extension) {
 
       // Else, add a full web3 instance
       if (!web3.injectedWeb3) {
-        const rawWeb3 = require('web3/dist/web3.min.js');
-        eval(rawWeb3); // eslint-disable-line no-eval
-
-        const injectedWeb3 = new window.Web3(web3.currentProvider);
-        web3.injectedWeb3 = injectedWeb3;
-        // Expose object directly
-        window.web3 = injectedWeb3;
-        return injectedWeb3[name];
+        require('./lib');
```

Comment:
So we inject `web3` object anyway (if injection is enabled).
The recommended way of initializing web3 is (https://github.com/ethereum/wiki/wiki/JavaScript-API#adding-web3)
```
if (typeof web3 !== 'undefined') {
  web3 = new Web3(web3.currentProvider);
} else {
  // set the provider you want from Web3.providers
  web3 = new Web3(new Web3.providers.HttpProvider("http://localhost:8545"));
}
```

So people can still instantiate their own `Web3` version (loading the script twice will override previous `window.Web3`), they should just re-use the provider, that's why initially we were just mocking `web3.currentProvider`.

Since it turned out that some devs are not aware of that and they are just using web3 object directly, it might be best to just expose such object.

The question now is: is there any gain by lazy-instantiating Web3 via Proxy if library is loaded and parsed anyway?